cli/cmd/vm_ssh.go

```diff
@@ -0,0 +1,144 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/manifoldco/promptui"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/replicated/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func (r *runners) InitVMSSH(parent *cobra.Command) *cobra.Command {
+	var sshUser string
+
+	cmd := &cobra.Command{
+		Use:   "ssh [VM_ID]",
+		Short: "SSH into a VM",
+		Long: `Connect to a VM using SSH.
+
+If a VM ID is provided, it will directly connect to that VM. Otherwise, if multiple VMs are available, you will be prompted to select the VM you want to connect to. The command will then establish an SSH connection to the selected VM using the appropriate credentials and configuration.
+
+The SSH user can be specified in order of precedence:
+1. By specifying the -u flag
+2. REPLICATED_SSH_USER environment variable
+3. GITHUB_ACTOR environment variable (from GitHub Actions)
+4. GITHUB_USER environment variable
+
+Note: Only running VMs can be connected to via SSH.`,
+		Example: `# SSH into a specific VM by ID
+replicated vm ssh <id>
+
+# SSH into a VM with a specific user
+replicated vm ssh <id> -u myuser
+
+# SSH into a VM (interactive selection if multiple VMs exist)
+replicated vm ssh`,
+		RunE:              r.sshVM,
+		Args:              cobra.MaximumNArgs(1),
+		ValidArgsFunction: r.completeVMIDs,
+	}
+	parent.AddCommand(cmd)
+
+	cmd.Flags().StringVarP(&sshUser, "user", "u", "", "SSH user to connect with")
+
+	return cmd
+}
+
+func (r *runners) sshVM(cmd *cobra.Command, args []string) error {
+	if err := r.initVMClient(); err != nil {
+		return err
+	}
+
+	sshUser, _ := cmd.Flags().GetString("user")
+
+	// If VM ID is provided, directly SSH into it
+	if len(args) == 1 {
+		vmID := args[0]
+
+		// Check VM status before connecting
+		vm, err := r.kotsAPI.GetVM(vmID)
+		if err != nil {
+			return errors.Wrap(err, "failed to get VM")
+		}
+
+		if vm.Status != types.VMStatusRunning {
```

Comment:
This check gets duplicated in https://github.com/replicatedhq/replicated/pull/512/files#diff-c575af5e04f2c78709e450752a0a55c0aa4b96e4507bea9013e98395f6be979aR24, it's probably not necessary in both places.  Perhaps simply `r.sshVM` to just get the flags from cobra and then `return r.kotsAPI.SSHIntoVM` and handle all of the validation there?

```go
func (r *runners) sshVM(cmd *cobra.Command, args []string) error {
	if err := r.initVMClient(); err != nil {
		return err
	}

	sshUser, _ := cmd.Flags().GetString("user")

        return r.kotsAPI.SSHIntoVM(vmID, sshUser)
}